Boltdir/site-modules/metrics/files/collect_metrics_files.rb

```diff
@@ -0,0 +1,377 @@
+    #   stage_json(file_path)
+    def stage_json(file_path)
+      filename = File.basename(file_path)
+      file_dir = File.dirname(file_path)
+      relative_dir_path = file_dir.gsub(%r{^#{@metrics_dir}/}, "")
+      destination_dir = "#{@staging_dir}/#{relative_dir_path}"
+      FileUtils.mkdir(destination_dir) unless File.directory?(destination_dir)
+      FileUtils.cp(file_path, "#{destination_dir}/#{filename}")
+    end
+
+    # Determines if a tar file contains jsons from the target window and if so,
+    # either copies the whole tar file, or extracts the jsons from it to copy the relevant ones to
+    # the output_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to tar file
+    # @param [string] service name of the service the file comes from
+    #
+    # @return [void]
+    #
+    # @example
+    #   process_tar_file(file_path, service)
+    def consider_tarfile_contents_for_inclusion(file_path, service)
+      filename = File.basename(file_path)
+      return unless tarfile_from_target_window?(filename, service)
+
+      # this is only here because without it the script doesn't output much and might appear to be hung
+      puts "Checking #{filename} for relevant jsons"
+      zr = Zlib::GzipReader.new(File.open(file_path, "rb"))
+      mtr = Archive::Tar::Minitar::Reader.new(zr)
+      jsons_to_include = []
+      mtr.each do |file|
+        filename = File.basename(file.full_name)
+        jsons_to_include << file.full_name if json_file_from_target_window?(filename)
+      end
+      return if jsons_to_include.empty?
+
+      service_staging_dir = "#{@staging_dir}/#{service}"
+      FileUtils.mkdir(service_staging_dir) unless File.directory?(service_staging_dir)
+      zr.rewind
+      Archive::Tar::Minitar.unpack(zr, service_staging_dir, jsons_to_include)
+    end
+
+    # Determines if the tarfile may contain json files from the target window
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] filename The tar filename
+    # @param [string] service name of the service the file comes from
+    #
+    # @return [boolean] true if yes, false if no
+    #
+    # @example
+    #   tarfile_from_target_window?(filename, service)
+    def tarfile_from_target_window?(filename, service)
+      time_str_with_extension = filename.gsub(/^#{service}-/, "")
+      time_str = time_str_with_extension.gsub(/.tar.gz$/, "")
+      time_obj = DateTime.strptime(time_str, "%Y.%m.%d.%H.%M.%S")
+      epoch = time_obj.to_time.to_i
+      return true if (epoch - @archive_interval) > @start_epoch && epoch < @end_epoch
+
+      return true if epoch > @start_epoch && (epoch - @archive_interval) < @end_epoch
+
+      return false
+    end
+
+    # Tars up the resulting metrics files for easy transport off the system
+    #
+    # @author Randell Pelak
+    #
+    # @return [void]
+    #
+    # @example
+    #   tar_metrics_files
+    def tar_metrics_files
+      v = (@verbose ? "v" : "")
```

Comment:
What is the point of this ternary call?  You use the `@verbose` variable below.

Reply:
The tar command uses the v variable.  This was to avoid duplicating the rest of the command in an if verbose block.  I feel like there is a better way to do it, but I couldn't think of one.

Reply:
Oh, I see it now. Thanks for the explanation.  I completely missed the variable embedded in the flag below.